attributes/alibaba.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+cred_dir = ::File.expand_path(::File.join('~', '.alibaba'))
+credential_file = ::File.join(cred_dir, 'credentials')
+
+TIMESTAMP = Time.now.strftime('%Y_%m_%d_%H_%M_%S').freeze
+INVALID_AMI_NAME_CHARACTERS = /[^\w\(\)\.\-\/]/
+
+def normalize_name(name)
+  # Append timestamp
+  # Remove invalid characters
+  # Ensure it's 128 characters or less
+  "#{name}_#{TIMESTAMP}".gsub(INVALID_AMI_NAME_CHARACTERS, '').strip[0..127]
+end
+
+default['marketplace_image']['alibaba']['cred_dir'] = cred_dir
+default['marketplace_image']['alibaba']['credential_file'] = credential_file
+
+default['marketplace_image']['alibaba']['public']['compliance']['enabled'] = false
+default['marketplace_image']['alibaba']['public']['automate']['enabled'] = false
+
+default['marketplace_image']['alibaba']['product_urls'] = {
+  'marketplace' => 'http://chef-software.oss-cn-beijing.aliyuncs.com/chef-marketplace_0.2.5%2B20170911103840.git.2.1c6a67f-1_amd64.deb',
+  'automate' => 'http://chef-software.oss-cn-beijing.aliyuncs.com/automate_1.6.95-1_amd64.deb',
+  'chef_server' => 'http://chef-software.oss-cn-beijing.aliyuncs.com/chef-server-core_12.15.8-1_amd64.deb',
+}
+
+default_marketplace_config = {
+  'role' => 'automate',
+  'platform' => 'alibaba',
+  'user' => 'ubuntu',
+  'support_email' => 'aws@chef.io',
+  'sales_email' => 'awesome@chef.io',
+  'reporting_cron_enabled' => false,
+  'doc_url' => 'https://docs.chef.io/aws_marketplace.html',
```

Comment:
we should probably make this specific to alibaba as well.

Reply:
Have changed this to an Alibaba specific address - again it needs creating.